Firestore/core/src/firebase/firestore/core/event_listener.h

```diff
@@ -132,6 +153,9 @@ void AsyncEventListener<T>::OnEvent(util::StatusOr<T> maybe_value) {
   std::shared_ptr<AsyncEventListener<T>> shared_this = this->shared_from_this();
 
   executor_->Execute([shared_this, maybe_value]() {
+    // Hold the lock while calling the delegate in order to prevent concurrent
```

Comment:
Perhaps add a note like (`to prevent concurrent destruction of the Firestore instance (which will block on trying to mute the listener)`)? Without context, it's not immediately clear what the lock has to do with destroying a Firestore instance.